README.md

```diff
@@ -108,10 +108,27 @@ If `swiftc` is not in your `PATH`, you must specify the path to it using
 `-D CMAKE_Swift_COMPILER=`.
 
 ```
-cmake -G Ninja -B out -S swift-apis
+cmake -B out -D USE_BUNDLED_CTENSORFLOW=YES -D BUILD_X10=YES -G Ninja -S swift-apis
 cmake --build out
 ```
 
+If you are intending to develop X10, we recommend that you pass the
+`-no-stdlib-rpath` flag during compilation:
+
+```
+cmake -B out -D BUILD_X10=YES -D USE_BUNDLED_CTENSORFLOW=YES -D CMAKE_Swift_FLAGS="-no-stdlib-rpath" -G Ninja -S swift-apis
```

Comment:
Hmmm, with a clean checkout of this repository using a relatively recent toolchain (details below), I'm not able to successfully build following this command:

```
saeta@saeta:~/src/s4tf3$ swift --version
Swift version 5.3-dev (LLVM 5a342fdfac, Swift 8bc7b675d8)
Target: x86_64-unknown-linux-gnu
```

```
[14/23] Linking Swift shared library Sources/x10/libx10_tensor.so
FAILED: Sources/x10/libx10_tensor.so Sources/x10/CMakeFiles/x10_tensor.dir/swift_bindings/apis/CrossReplicaSum.swift.o Sources/x10/CMakeFiles/x10_tensor.dir/swift_bindings/apis/DeviceScope.swift.o Sources/x10/CMakeFiles/
x10_tensor.dir/swift_bindings/apis/RawOpsManual.swift.o Sources/x10/CMakeFiles/x10_tensor.dir/swift_bindings/Device.swift.o Sources/x10/CMakeFiles/x10_tensor.dir/swift_bindings/XLAScalarType.swift.o Sources/x10/CMakeFile
s/x10_tensor.dir/swift_bindings/XLATensor.swift.o Sources/x10/CMakeFiles/x10_tensor.dir/__/TensorFlow/Bindings/EagerExecution.swift.o Sources/x10/CMakeFiles/x10_tensor.dir/__/TensorFlow/Bindings/RawOpsGenerated.swift.o S
ources/x10/CMakeFiles/x10_tensor.dir/__/TensorFlow/Bindings/RawOpsAugmented.swift.o Sources/x10/CMakeFiles/x10_tensor.dir/__/TensorFlow/Bindings/RawOpsDispatching.swift.o Sources/x10/CMakeFiles/x10_tensor.dir/__/TensorFl
ow/Bindings/TFTensorOperation.swift.o Sources/x10/CMakeFiles/x10_tensor.dir/__/TensorFlow/Core/BroadcastingPullback.swift.o Sources/x10/CMakeFiles/x10_tensor.dir/__/TensorFlow/Core/CopyableToDevice.swift.o Sources/x10/CM
akeFiles/x10_tensor.dir/__/TensorFlow/Core/DifferentialOperators.swift.o Sources/x10/CMakeFiles/x10_tensor.dir/__/TensorFlow/Core/PythonConversion.swift.o Sources/x10/CMakeFiles/x10_tensor.dir/__/TensorFlow/Core/ShapedAr
ray.swift.o Sources/x10/CMakeFiles/x10_tensor.dir/__/TensorFlow/Core/Execution.swift.o Sources/x10/CMakeFiles/x10_tensor.dir/__/TensorFlow/Core/TensorShape.swift.o Sources/x10/CMakeFiles/x10_tensor.dir/__/TensorFlow/Core
/Threading.swift.o Sources/x10/CMakeFiles/x10_tensor.dir/__/TensorFlow/Core/Utilities.swift.o Sources/x10/CMakeFiles/x10_tensor.dir/__/TensorFlow/Core/Tensor.swift.o Sources/x10/CMakeFiles/x10_tensor.dir/__/TensorFlow/Co
re/TensorHandle.swift.o Sources/x10/CMakeFiles/x10_tensor.dir/__/TensorFlow/Core/DataTypes.swift.o Sources/x10/CMakeFiles/x10_tensor.dir/__/TensorFlow/Core/MixedPrecision.swift.o Sources/x10/CMakeFiles/x10_tensor.dir/__/
TensorFlow/Core/Runtime.swift.o Sources/x10/CMakeFiles/x10_tensor.dir/__/TensorFlow/Core/StringTensor.swift.o Sources/x10/CMakeFiles/x10_tensor.dir/__/TensorFlow/Core/TensorGroup.swift.o Sources/x10/CMakeFiles/x10_tensor
.dir/__/TensorFlow/Core/ArrayOps.swift.o Sources/x10/CMakeFiles/x10_tensor.dir/__/TensorFlow/Core/LazyTensorContext.swift.o Sources/x10/CMakeFiles/x10_tensor.dir/__/TensorFlow/Core/LazyTensorOperation.swift.o Sources/x10
/CMakeFiles/x10_tensor.dir/__/TensorFlow/Core/LazyTensorShapeInference.swift.o Sources/x10/CMakeFiles/x10_tensor.dir/__/TensorFlow/Core/LazyTensorTFFunctionBuilder.swift.o Sources/x10/CMakeFiles/x10_tensor.dir/__/TensorF
low/Core/LazyTensorTrace.swift.o Sources/x10/CMakeFiles/x10_tensor.dir/__/TensorFlow/Core/LazyTensorTraceCache.swift.o Sources/x10/CMakeFiles/x10_tensor.dir/__/TensorFlow/Core/Serialization.swift.o Sources/x10/CMakeFiles
/x10_tensor.dir/__/TensorFlow/Layers/Convolutional.swift.o Sources/x10/CMakeFiles/x10_tensor.dir/__/TensorFlow/Layers/Core.swift.o Sources/x10/CMakeFiles/x10_tensor.dir/__/TensorFlow/Layers/Dense.swift.o Sources/x10/CMak
eFiles/x10_tensor.dir/__/TensorFlow/Layers/Dropout.swift.o Sources/x10/CMakeFiles/x10_tensor.dir/__/TensorFlow/Layers/Embedding.swift.o Sources/x10/CMakeFiles/x10_tensor.dir/__/TensorFlow/Layers/Initialization.swift.o So
urces/x10/CMakeFiles/x10_tensor.dir/__/TensorFlow/Layers/Normalization.swift.o Sources/x10/CMakeFiles/x10_tensor.dir/__/TensorFlow/Layers/Pooling.swift.o Sources/x10/CMakeFiles/x10_tensor.dir/__/TensorFlow/Layers/Recurre
nt.swift.o Sources/x10/CMakeFiles/x10_tensor.dir/__/TensorFlow/Layers/Sequential.swift.o Sources/x10/CMakeFiles/x10_tensor.dir/__/TensorFlow/Layers/Upsampling.swift.o Sources/x10/CMakeFiles/x10_tensor.dir/__/TensorFlow/O
perators/Basic.swift.o Sources/x10/CMakeFiles/x10_tensor.dir/__/TensorFlow/Operators/Comparison.swift.o Sources/x10/CMakeFiles/x10_tensor.dir/__/TensorFlow/Operators/Dataset.swift.o Sources/x10/CMakeFiles/x10_tensor.dir/
__/TensorFlow/Operators/Image.swift.o Sources/x10/CMakeFiles/x10_tensor.dir/__/TensorFlow/Operators/LinearAlgebra.swift.o Sources/x10/CMakeFiles/x10_tensor.dir/__/TensorFlow/Operators/Math.swift.o Sources/x10/CMakeFiles/
x10_tensor.dir/__/TensorFlow/Operators/NN.swift.o Sources/x10/CMakeFiles/x10_tensor.dir/__/TensorFlow/Optimizers/MomentumBased.swift.o Sources/x10/CMakeFiles/x10_tensor.dir/__/TensorFlow/Optimizers/Optimizer.swift.o Sour
ces/x10/CMakeFiles/x10_tensor.dir/__/TensorFlow/Optimizers/SGD.swift.o Sources/x10/CMakeFiles/x10_tensor.dir/__/TensorFlow/BackwardsCompatibility.swift.o Sources/x10/CMakeFiles/x10_tensor.dir/__/TensorFlow/Context.swift.
o Sources/x10/CMakeFiles/x10_tensor.dir/__/TensorFlow/Exports.swift.o Sources/x10/CMakeFiles/x10_tensor.dir/__/TensorFlow/Freezable.swift.o Sources/x10/CMakeFiles/x10_tensor.dir/__/TensorFlow/Initializers.swift.o Sources
/x10/CMakeFiles/x10_tensor.dir/__/TensorFlow/Layer.swift.o Sources/x10/CMakeFiles/x10_tensor.dir/__/TensorFlow/Loss.swift.o Sources/x10/CMakeFiles/x10_tensor.dir/__/TensorFlow/StdlibExtensions.swift.o swift/x10_tensor.sw
iftmodule 
: && /usr/local/google/home/saeta/src/s4tf3/swift/swift-nightly-install/usr/bin/swiftc -output-file-map Sources/x10/CMakeFiles/x10_tensor.dir//output-file-map.json -incremental -j 72 -emit-library -o Sources/x10/libx10_t
ensor.so -module-name x10_tensor -module-link-name x10_tensor -emit-module -emit-module-path swift/x10_tensor.swiftmodule -emit-dependencies -DUSING_X10_BACKEND -Dx10_tensor_EXPORTS -no-stdlib-rpath   -Xllvm -sil-inline-
generics -Xllvm -sil-partial-specialization -I /usr/local/google/home/saeta/src/s4tf3/tensorflow-swift-apis/Sources/x10 -I swift -I libtensorflow-prefix/src/libtensorflow -I libtensorflow-prefix/src/libtensorflow/bazel-b
in -I libtensorflow-prefix/src/libtensorflow/bazel-libtensorflow/external/com_google_absl -I libtensorflow-prefix/src/libtensorflow/bazel-libtensorflow/external/com_google_protobuf/src -I libtensorflow-prefix/src/libtens
orflow/bazel-libtensorflow/external/eigen_archive /usr/local/google/home/saeta/src/s4tf3/tensorflow-swift-apis/Sources/x10/swift_bindings/apis/CrossReplicaSum.swift /usr/local/google/home/saeta/src/s4tf3/tensorflow-swift
-apis/Sources/x10/swift_bindings/apis/DeviceScope.swift /usr/local/google/home/saeta/src/s4tf3/tensorflow-swift-apis/Sources/x10/swift_bindings/apis/RawOpsManual.swift /usr/local/google/home/saeta/src/s4tf3/tensorflow-sw
ift-apis/Sources/x10/swift_bindings/Device.swift /usr/local/google/home/saeta/src/s4tf3/tensorflow-swift-apis/Sources/x10/swift_bindings/XLAScalarType.swift /usr/local/google/home/saeta/src/s4tf3/tensorflow-swift-apis/So
urces/x10/swift_bindings/XLATensor.swift /usr/local/google/home/saeta/src/s4tf3/tensorflow-swift-apis/Sources/TensorFlow/Bindings/EagerExecution.swift /usr/local/google/home/saeta/src/s4tf3/tensorflow-swift-apis/Sources/
TensorFlow/Bindings/RawOpsGenerated.swift /usr/local/google/home/saeta/src/s4tf3/tensorflow-swift-apis/Sources/TensorFlow/Bindings/RawOpsAugmented.swift /usr/local/google/home/saeta/src/s4tf3/tensorflow-swift-apis/Source
s/TensorFlow/Bindings/RawOpsDispatching.swift /usr/local/google/home/saeta/src/s4tf3/tensorflow-swift-apis/Sources/TensorFlow/Bindings/TFTensorOperation.swift /usr/local/google/home/saeta/src/s4tf3/tensorflow-swift-apis/
Sources/TensorFlow/Core/BroadcastingPullback.swift /usr/local/google/home/saeta/src/s4tf3/tensorflow-swift-apis/Sources/TensorFlow/Core/CopyableToDevice.swift /usr/local/google/home/saeta/src/s4tf3/tensorflow-swift-apis/
Sources/TensorFlow/Core/DifferentialOperators.swift /usr/local/google/home/saeta/src/s4tf3/tensorflow-swift-apis/Sources/TensorFlow/Core/PythonConversion.swift /usr/local/google/home/saeta/src/s4tf3/tensorflow-swift-apis
/Sources/TensorFlow/Core/ShapedArray.swift /usr/local/google/home/saeta/src/s4tf3/tensorflow-swift-apis/Sources/TensorFlow/Core/Execution.swift /usr/local/google/home/saeta/src/s4tf3/tensorflow-swift-apis/Sources/TensorF
low/Core/TensorShape.swift /usr/local/google/home/saeta/src/s4tf3/tensorflow-swift-apis/Sources/TensorFlow/Core/Threading.swift /usr/local/google/home/saeta/src/s4tf3/tensorflow-swift-apis/Sources/TensorFlow/Core/Utiliti
es.swift /usr/local/google/home/saeta/src/s4tf3/tensorflow-swift-apis/Sources/TensorFlow/Core/Tensor.swift /usr/local/google/home/saeta/src/s4tf3/tensorflow-swift-apis/Sources/TensorFlow/Core/TensorHandle.swift /usr/loca
l/google/home/saeta/src/s4tf3/tensorflow-swift-apis/Sources/TensorFlow/Core/DataTypes.swift /usr/local/google/home/saeta/src/s4tf3/tensorflow-swift-apis/Sources/TensorFlow/Core/MixedPrecision.swift /usr/local/google/home
/saeta/src/s4tf3/tensorflow-swift-apis/Sources/TensorFlow/Core/Runtime.swift /usr/local/google/home/saeta/src/s4tf3/tensorflow-swift-apis/Sources/TensorFlow/Core/StringTensor.swift /usr/local/google/home/saeta/src/s4tf3/
tensorflow-swift-apis/Sources/TensorFlow/Core/TensorGroup.swift /usr/local/google/home/saeta/src/s4tf3/tensorflow-swift-apis/Sources/TensorFlow/Core/ArrayOps.swift /usr/local/google/home/saeta/src/s4tf3/tensorflow-swift-
apis/Sources/TensorFlow/Core/LazyTensorContext.swift /usr/local/google/home/saeta/src/s4tf3/tensorflow-swift-apis/Sources/TensorFlow/Core/LazyTensorOperation.swift /usr/local/google/home/saeta/src/s4tf3/tensorflow-swift-
apis/Sources/TensorFlow/Core/LazyTensorShapeInference.swift /usr/local/google/home/saeta/src/s4tf3/tensorflow-swift-apis/Sources/TensorFlow/Core/LazyTensorTFFunctionBuilder.swift /usr/local/google/home/saeta/src/s4tf3/te
nsorflow-swift-apis/Sources/TensorFlow/Core/LazyTensorTrace.swift /usr/local/google/home/saeta/src/s4tf3/tensorflow-swift-apis/Sources/TensorFlow/Core/LazyTensorTraceCache.swift /usr/local/google/home/saeta/src/s4tf3/ten
sorflow-swift-apis/Sources/TensorFlow/Core/Serialization.swift /usr/local/google/home/saeta/src/s4tf3/tensorflow-swift-apis/Sources/TensorFlow/Layers/Convolutional.swift /usr/local/google/home/saeta/src/s4tf3/tensorflow-
swift-apis/Sources/TensorFlow/Layers/Core.swift /usr/local/google/home/saeta/src/s4tf3/tensorflow-swift-apis/Sources/TensorFlow/Layers/Dense.swift /usr/local/google/home/saeta/src/s4tf3/tensorflow-swift-apis/Sources/Tens
orFlow/Layers/Dropout.swift /usr/local/google/home/saeta/src/s4tf3/tensorflow-swift-apis/Sources/TensorFlow/Layers/Embedding.swift /usr/local/google/home/saeta/src/s4tf3/tensorflow-swift-apis/Sources/TensorFlow/Layers/In
itialization.swift /usr/local/google/home/saeta/src/s4tf3/tensorflow-swift-apis/Sources/TensorFlow/Layers/Normalization.swift /usr/local/google/home/saeta/src/s4tf3/tensorflow-swift-apis/Sources/TensorFlow/Layers/Pooling
.swift /usr/local/google/home/saeta/src/s4tf3/tensorflow-swift-apis/Sources/TensorFlow/Layers/Recurrent.swift /usr/local/google/home/saeta/src/s4tf3/tensorflow-swift-apis/Sources/TensorFlow/Layers/Sequential.swift /usr/l
ocal/google/home/saeta/src/s4tf3/tensorflow-swift-apis/Sources/TensorFlow/Layers/Upsampling.swift /usr/local/google/home/saeta/src/s4tf3/tensorflow-swift-apis/Sources/TensorFlow/Operators/Basic.swift /usr/local/google/ho
me/saeta/src/s4tf3/tensorflow-swift-apis/Sources/TensorFlow/Operators/Comparison.swift /usr/local/google/home/saeta/src/s4tf3/tensorflow-swift-apis/Sources/TensorFlow/Operators/Dataset.swift /usr/local/google/home/saeta/
src/s4tf3/tensorflow-swift-apis/Sources/TensorFlow/Operators/Image.swift /usr/local/google/home/saeta/src/s4tf3/tensorflow-swift-apis/Sources/TensorFlow/Operators/LinearAlgebra.swift /usr/local/google/home/saeta/src/s4tf
3/tensorflow-swift-apis/Sources/TensorFlow/Operators/Math.swift /usr/local/google/home/saeta/src/s4tf3/tensorflow-swift-apis/Sources/TensorFlow/Operators/NN.swift /usr/local/google/home/saeta/src/s4tf3/tensorflow-swift-a
pis/Sources/TensorFlow/Optimizers/MomentumBased.swift /usr/local/google/home/saeta/src/s4tf3/tensorflow-swift-apis/Sources/TensorFlow/Optimizers/Optimizer.swift /usr/local/google/home/saeta/src/s4tf3/tensorflow-swift-api
s/Sources/TensorFlow/Optimizers/SGD.swift /usr/local/google/home/saeta/src/s4tf3/tensorflow-swift-apis/Sources/TensorFlow/BackwardsCompatibility.swift /usr/local/google/home/saeta/src/s4tf3/tensorflow-swift-apis/Sources/
TensorFlow/Context.swift /usr/local/google/home/saeta/src/s4tf3/tensorflow-swift-apis/Sources/TensorFlow/Exports.swift /usr/local/google/home/saeta/src/s4tf3/tensorflow-swift-apis/Sources/TensorFlow/Freezable.swift /usr/
local/google/home/saeta/src/s4tf3/tensorflow-swift-apis/Sources/TensorFlow/Initializers.swift /usr/local/google/home/saeta/src/s4tf3/tensorflow-swift-apis/Sources/TensorFlow/Layer.swift /usr/local/google/home/saeta/src/s
4tf3/tensorflow-swift-apis/Sources/TensorFlow/Loss.swift /usr/local/google/home/saeta/src/s4tf3/tensorflow-swift-apis/Sources/TensorFlow/StdlibExtensions.swift  -Xlinker -soname -Xlinker libx10_tensor.so  -L /usr/local/g
oogle/home/saeta/src/s4tf3/out/Sources/Tensor   -L /usr/local/google/home/saeta/src/s4tf3/out/libtensorflow-prefix/src/libtensorflow/bazel-bin/tensorflow/compiler/tf2xla/xla_tensor -Xlinker -rpath -Xlinker /usr/local/goo
gle/home/saeta/src/s4tf3/out/libtensorflow-prefix/src/libtensorflow/bazel-bin/tensorflow/compiler/tf2xla/xla_tensor:/usr/local/google/home/saeta/src/s4tf3/out/Sources/Tensor:  Sources/Tensor/libTensor.so  libtensorflow-p
refix/src/libtensorflow/bazel-bin/tensorflow/compiler/tf2xla/xla_tensor/libx10.so && :
/usr/local/google/home/saeta/src/s4tf3/tensorflow-swift-apis/Sources/TensorFlow/Bindings/EagerExecution.swift:17:8: error: no such module 'CTensorFlow'
import CTensorFlow
       ^
ninja: build stopped: subcommand failed.
```

Any hints?

Reply:
Note: I also get a failure when trying to build without x10 and get a similar error:

```
cmake -B out -D USE_BUNDLED_CTENSORFLOW=YES -D BUILD_X10=YES -G Ninja -S tensorflow-swift-apis
cmake --build out
```

Thanks!

Reply:
The nightlies haven't been updated yet, these require toolchains not yet available.  But, I didn't want to forget about the documentation since @pschuh was surprised about the behaviour, I figured that if he was caught surprised most of us don't stand a chance.